packages/sandcastle/src/Gallery/loadFromUrl.ts

```diff
@@ -0,0 +1,103 @@
+import { type GalleryItem } from "./GalleryItemStore.ts";
+import { decodeBase64Data } from "../Helpers.ts";
+import { getBaseUrl } from "../util/getBaseUrl.ts";
+
+const loadGist = async (gist: string) => {
+  try {
+    const response = await fetch(`https://api.github.com/gists/${gist}`);
+    const data = await response.json();
+    const files = data.files;
+    if (!files) {
+      throw new Error(data.message);
+    }
+    const code = files["Cesium-Sandcastle.js"]?.content;
+    const html = files["Cesium-Sandcastle.html"]?.content;
+    return {
+      title: "Gist Import",
+      code,
+      html,
+    };
+  } catch (error) {
+    throw new Error(
+      `${error}. Unable to requets gist from GitHub. This could be due to too many requests from your IP or an incorrect gist ID. Instead, copy and paste the code from "https://gist.github.com/${gist}"`,
+    );
+  }
+};
+
+const fromItem = async ({ getHtmlCode, getJsCode, title }: GalleryItem) => {
+  try {
+    const [code, html] = await Promise.all([getJsCode(), getHtmlCode()]);
+    return {
+      title,
+      code,
+      html,
+    };
+  } catch (error) {
+    throw new Error(`Could not load "${title}": ${error}`);
+  }
+};
+
+export function loadFromUrl(
+  items: GalleryItem[],
+  legacyIds: Record<string, string>,
+) {
+  const searchParams = new URLSearchParams(window.location.search);
+
+  if (window.location.hash.indexOf("#c=") === 0) {
+    const base64String = window.location.hash.substr(3);
+    const { code, html } = decodeBase64Data(base64String);
+    return {
+      title: "Sandcaste",
```

Comment:
```suggestion
      title: "Sandcastle",
```

Also setting this results in a title of "Sandcaste | Sandcastle | CesiumJS" which is very redundant. The previous behavior was all sandcastles would have the title "New sandcastle" if they weren't a specific demo. If we want to depart from that I'm open to it but the title logic should probably update accordingly to prevent the double "Sandcastle"